### PR TITLE
Fixed an issue with merged 1-4 interactions

### PR DESCRIPTION
**Pull request by msuruzhon**
This PR fixes an issue observed after the recent changes in the merge function, where the outputted 1-4 interactions were wrong.

### CODE REVIEW

File: python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py [hunk 1]
```diff
@@ -959,18 +959,10 @@ def merge(molecule0, molecule1, mapping, allow_ring_breaking=False,
...
-        clj_nb_pairs1 = _SireMM.CLJNBPairs(edit_mol.info(),
-                                           _SireMM.CLJScaleFactor(0, 0))
-    # When merging proteins, initilize the clj_nb_pairs from the
-    # connection information of molecule0 and molecule1
-    else:
-        scale_factor_14 = _SireMM.CLJScaleFactor(ff.electrostatic14ScaleFactor(),
-                                                 ff.vdw14ScaleFactor())
-        clj_nb_pairs0 = _SireMM.CLJNBPairs(conn0, scale_factor_14)
-        clj_nb_pairs1 = _SireMM.CLJNBPairs(conn1, scale_factor_14)
+    scale_factor_14 = _SireMM.CLJScaleFactor(ff.electrostatic14ScaleFactor(),
```
[the remark this answers is not in the document]
Comment (xiki-tempula) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: What about roi then? Do you not treat them separately?

[the remark this answers is not in the document]
Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: I didn't get that issue with `roi`, only when it was `None`, so I replaced the non-roi implementation with this. Seems fine for now.

[the remark this answers is not in the document]
Comment (xiki-tempula) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: Can you add a simple test to show that this is fixed? Just show that a pair of interaction is in the topology is fine.

[the remark this answers is not in the document]
Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: I have a test that counts the number of pairs. Is this okay?

[the remark this answers is not in the document]
Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: No but that's a test I added today that I made sure was broken before the fix

[the remark this answers is not in the document]
Comment (xiki-tempula) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_merge.py:962]: Oh, weird that I cannot see the test before.

**Review by xiki-tempula — APPROVED** [review]
LGTM

**Pull request merged**